Frameworks/CoreText/CTParagraphStyle.mm

```diff
@@ -17,235 +17,182 @@
-            {
-                sizeof(CGFloat), offsetof(_CTParagraphStyleProperties, _paragraphSpacingBefore),
-            },
-        [kCTParagraphStyleSpecifierBaseWritingDirection] =
-            {
-                sizeof(CTWritingDirection), offsetof(_CTParagraphStyleProperties, _writingDirection),
-            },
-        [kCTParagraphStyleSpecifierMaximumLineSpacing] =
-            {
-                sizeof(CGFloat), offsetof(_CTParagraphStyleProperties, _maximumLineSpacing),
-            },
-        [kCTParagraphStyleSpecifierMinimumLineSpacing] =
-            {
-                sizeof(CGFloat), offsetof(_CTParagraphStyleProperties, _minimumLineSpacing),
-            },
-        [kCTParagraphStyleSpecifierLineSpacingAdjustment] =
-            {
-                sizeof(CGFloat), offsetof(_CTParagraphStyleProperties, _lineSpacingAdjustment),
-            },
+#pragma region __CTParagraphStyle
+typedef struct __CTParagraphStyle : CoreFoundation::CppBase<__CTParagraphStyle> {
+    __CTParagraphStyle() {
+        static CFArrayRef __defaultValues = []() {
+            constexpr CGFloat c_defaultFloat = 0.0;
+            // Need to store Alignment and LineBreakMode as ints in CFNumberRef
+            constexpr int c_defaultAlignment = kCTNaturalTextAlignment;
+            constexpr int c_defaultLineBreakMode = kCTLineBreakByWordWrapping;
+            constexpr CTWritingDirection c_defaultWritingDirection = kCTWritingDirectionNatural;
+            woc::unique_cf<CFNumberRef> __alignment{ CFNumberCreate(nullptr, kCFNumberIntType, &c_defaultAlignment) };
+            woc::unique_cf<CFNumberRef> __lineBreakMode{ CFNumberCreate(nullptr, kCFNumberIntType, &c_defaultLineBreakMode) };
+            woc::unique_cf<CFNumberRef> __writingDirection{ CFNumberCreate(nullptr, kCFNumberSInt8Type, &c_defaultWritingDirection) };
+            woc::unique_cf<CFNumberRef> __floatDefault{ CFNumberCreate(nullptr, kCFNumberCGFloatType, &c_defaultFloat) };
+            CFTypeRef values[kCTParagraphStyleSpecifierCount] = {
+                __alignment.get(),    __floatDefault.get(),  __floatDefault.get(), __floatDefault.get(),     nullptr,
+                __floatDefault.get(), __lineBreakMode.get(), __floatDefault.get(), __floatDefault.get(),     __floatDefault.get(),
+                __floatDefault.get(), __floatDefault.get(),  __floatDefault.get(), __writingDirection.get(), __floatDefault.get(),
+                __floatDefault.get(), __floatDefault.get(),
+            };
+            return CFArrayCreate(nullptr, values, kCTParagraphStyleSpecifierCount, &kCFTypeArrayCallBacks);
+        }();
+        _styles.reset(CFArrayCreateMutableCopy(nullptr, kCTParagraphStyleSpecifierCount, __defaultValues));
+    }
 
-};
+    __CTParagraphStyle(const __CTParagraphStyle& rhs) {
+        if (this != &rhs) {
+            _styles.reset(CFArrayCreateMutableCopy(nullptr, kCTParagraphStyleSpecifierCount, rhs._styles.get()));
```

Comment:
we should seek to declutter our code of unnecessary conditionals the compiler can easily optimize out in every case ;)

---
In reply to: [87272321](https://github.com/Microsoft/WinObjC/pull/1343#discussion_r87272321) [](ancestors = 87272321)